onnxruntime/core/providers/cpu/ml/tree_ensemble_aggregator.h

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "ml_common.h"
+#include <math.h>
+
+namespace onnxruntime {
+namespace ml {
+
+struct TreeNodeElementId {
+  int tree_id;
+  int node_id;
+  inline bool operator==(const TreeNodeElementId& xyz) const {
+    return (tree_id == xyz.tree_id) && (node_id == xyz.node_id);
+  }
+  inline bool operator<(const TreeNodeElementId& xyz) const {
+    return ((tree_id < xyz.tree_id) || (tree_id == xyz.tree_id && node_id < xyz.node_id));
+  }
+};
+
+template <typename T>
+struct SparseValue {
+  int64_t i;
+  T value;
+};
+
+enum MissingTrack {
+  NONE,
+  TRUE,
+  FALSE
+};
+
+template <typename T>
+struct TreeNodeElement {
+  TreeNodeElementId id;
+  int feature_id;
+  T value;
+  T hitrates;
+  NODE_MODE mode;
+  TreeNodeElement<T>* truenode;
+  TreeNodeElement<T>* falsenode;
+  MissingTrack missing_tracks;
+  std::vector<SparseValue<T>> weights;
+
+  bool is_not_leave;
+  bool is_missing_track_true;
+};
+
+template <typename ITYPE, typename OTYPE>
+class _Aggregator {
+ protected:
```

Comment:
Make this non-copyable to prevent accidental copies. Use ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE or one of the lesser restrictive options as appropriate.

Also needs virtual destructor.

Reply:
destructor was removed (not needed anymore).